smaily-for-wordpress.php

```diff
@@ -233,19 +244,31 @@ function smaily_nojs_subscribe_callback() {
 
 	$array = array_merge( $array, $form_values );
 	require_once( BP . DS . 'code' . DS . 'Request.php' );
-	$request = new Wp_Sendsmaily_Request( $server, $array );
+	$request = new Wp_Smaily_Request( $server, $array );
 	$result  = $request->exec();
 
 	if ( empty( $result ) ) {
-		echo esc_html__( 'Something went wrong', 'wp_sendsmaily' );
+		$redirect_url .= '?smaily_form_error=' . esc_html__( 'Something went wrong', 'wp_smaily' );
 	} elseif ( (int) $result['code'] !== 101 ) {
-		// Possible errors, for translation.
-		// esc_html__('Posted fields do not contain a valid email address.', 'wp_sendsmaily');.
-		// esc_html__('No autoresponder data set.', 'wp_sendsmaily');.
-		echo esc_html__( $result['message'], 'wp_sendsmaily' );
+		switch ( $result['code'] ) {
+			case 201:
+				$error .= '?smaily_form_error=' . esc_html__( 'Form was not sent using POST method.', 'wp_smaily' );
+				break;
+			case 204:
+				$redirect_url .= '?smaily_form_error=' . esc_html__( 'Input does not contain a recognizable email address.', 'wp_smaily' );
+				break;
+			case 205:
+				$redirect_url .= '?smaily_form_error=' . esc_html__( 'Could not add to subscriber list for an unknown reason. Probably something in Smaily.', 'wp_smaily' );
+				break;
+			default:
+				$redirect_url .= '?smaily_form_error=' . esc_html__( 'Something went wrong', 'wp_smaily' );
+				break;
+		}
+		$redirect_url = str_replace(' ', '%20', $redirect_url);
+		wp_safe_redirect( $redirect_url );
+		exit;
 	}
-
-	wp_safe_redirect( home_url() );
+	wp_safe_redirect( remove_query_arg( 'smaily_form_error', $redirect_url ) );
```

Comment:
Why are you removing argument only here? What happens if URL contains `smaily_form_error` and nonce, posted data or email address validation fails?

Reply:
Then the URL and <p> will keep getting new arguments. Would a good solution be to move it after $redirect_url definition and write it like such
`$redirect_url = 
$referer_url ? remove_query_arg( 'smaily_form_error', $referer_url ) : get_home_url();`
or is always running remove_query_arg, regardless if it's present, bad?

Reply:
I would say it is the best solution :)

Always running `remove_query_arg` is not bad, because you are normalizing/cleaning up your input data, i.e. you want the referrer URL, but is should never contain `smaily_form_error` query argument. It would become bad, if `remove_query_arg` is a heavy operation that takes a long time.